doc/md/guides/orms/beego.md

```diff
@@ -0,0 +1,305 @@
+Beego provides an ORM that allows work with popular databases such as MySQL, PostgreSQL, SQLite3, and more.
+
+Beego allows developers to manage their database schemas using its [syncdb](https://github.com/beego/beedoc/blob/master/en-US/mvc/model/cmd.md#database-schema-generation)
+feature, which is usually sufficient during development and in many simple cases. 
+
+However, at some point, teams need more control and decide to employ 
+the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) 
+methodology. Once this happens, the responsibility for planning migration scripts and making
+sure they are in line with what Beego expects at runtime is moved to developers.
+
+Atlas can automatically plan database schema migrations for developers using Beego.
+Atlas plans migrations by calculating the diff between the _current_ state of the database,
+and its _desired_ state.
+
+In the context of versioned migrations, the current state can be thought of as the database schema that would have
+been created by applying all previous migration scripts.
+
+The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema)
+which is any program that can output a SQL schema definition to stdout.
+
+To use Atlas with Beego, users can utilize the [Beego Atlas Provider](https://github.com/ariga/atlas-provider-beego),
+a small Go program that can be used to load the schema of a Beego project into Atlas.
+
+In this guide, we will show how Atlas can be used to automatically plan schema migrations for
+Beego users.
+
+## Prerequisites
+
+* A local [Beego](https://github.com/beego/beego) project.
+
+## Using the Beego Atlas Provider 
+
+In this guide, we will use the [Beego Atlas Provider](https://github.com/ariga/atlas-provider-beego)
+to automatically plan schema migrations for a Beego project.
+
+### Installation
+
+Install Atlas from macOS or Linux by running:
+```bash
+curl -sSf https://atlasgo.sh | sh
+```
+See [atlasgo.io](https://atlasgo.io/getting-started#installation) for more installation options.
+
+Install the provider by running:
+```bash
+go get -u ariga.io/atlas-provider-beego
+``` 
+
+### Standalone vs Go Program mode
+
+The Atlas Beego Provider can be used in two modes:
+* **Standalone** - If your application contains a package that registers all of its Beego models 
+  during initialization (using an `func init()` function), you can use the provider directly to load your Beego schema into Atlas.
+* **Go Program** - In other cases, you can use the provider as a library directly in a Go program to load your Beego schema into Atlas.
+
+### Standalone mode
+
+In your project directory, create a new file named `atlas.hcl` with the following contents:
+
+```hcl
+data "external_schema" "beego" {
+  program = [
+    "go",
+    "run",
+    "-mod=mod",
+    "ariga.io/atlas-provider-beego",
```

Comment:
If we install the binary on the system, then we just need the program name to run.
```suggestion
    "atlas-provider-beego",
```